src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1461,8 +1470,14 @@ private AuthenticationResult localFlow(CallbackHandler callbackHandle,
                 // User does not want to launch activity
                 Logger.e(TAG, "Prompt is not allowed and failed to get token:", request.getLogInfo() + " " + errorInfo,
                         ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED);
-                callbackHandle.onError(new AuthenticationException(
-                        ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED, request.getLogInfo() + " " + errorInfo));
+                if (authResult != null && authResult.getErrorCode() != null) {
+                    callbackHandle.onError(new AuthenticationException(ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED, 
```

Comment:
The design here is, we parse the response from server, and if server returns us detailed error code, we propagated back to user.